### PR TITLE
bpo-30897: Add is_mount() to pathlib.Path

### PR DESCRIPTION
**Pull request by cooperlees**
* Add in is_mount() call to pathlib.Path similiar to os.path.ismount(path)
* Add tests for is_mount()

https://bugs.python.org/issue30897

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by cooperlees** [comment]
Getting a Windows VM to play with the error. All suggestions welcome!

**Comment by vstinner** [comment]
I removed "bpo-30891: " from the title, since http://bugs.python.org/issue30891 is unrelated.

File: Lib/pathlib.py [hunk 1]
```diff
@@ -1329,6 +1329,27 @@ def is_file(self):
             # (see https://bitbucket.org/pitrou/pathlib/issue/12/)
             return False
 
+    def is_mount(self):
+        """
```
Comment (mlouielu) [inline Lib/pathlib.py:1333]: would it be much easy to just return `os.path.ismount(self)` ?

Comment (cooperlees) [reply Lib/pathlib.py:1333]: Easier, possibly. But I still think I'll hit my current issue with Windows. No like function in the library hits the 'os' module and I am pretty sure Python core developers would want this library standalone where possible.

**Comment by cooperlees** [comment]
I may need to rebase - that appveyor failure looks unrelated to me.

**Comment by cooperlees** [comment]
Rebase got all checks to pass 💃

**Pull request merged**